From 2ea16c161ea4979b2bb96b31c8d9ceb3c6892a87 Mon Sep 17 00:00:00 2001
From: "Ye.Li" <B37916@freescale.com>
Date: Thu, 29 Jan 2015 13:51:51 +0800
Subject: [PATCH] MLK-10178-6 net:fec_mxc support shared MDIO bus

There are two FEC controller on MX7, and each MDIO bus can be shared
with other. So add a configuration for this case.

Signed-off-by: Peng Fan <Peng.Fan@freescale.com>
Signed-off-by: Ye.Li <B37916@freescale.com>
(cherry picked from commit d31b4fcc6175b1bc9a2cf672678da212b133af17)
---
 drivers/net/fec_mxc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/fec_mxc.c b/drivers/net/fec_mxc.c
index b572470..1c65bd7 100644
--- a/drivers/net/fec_mxc.c
+++ b/drivers/net/fec_mxc.c
@@ -1093,8 +1093,12 @@ int fecmxc_initialize_multi(bd_t *bd, int dev_id, int phy_id, uint32_t addr)
 	 */
 	base_mii = MXS_ENET0_BASE;
 #else
+#ifdef CONFIG_FEC_MXC_MDIO_BASE
+	base_mii = CONFIG_FEC_MXC_MDIO_BASE;
+#else
 	base_mii = addr;
 #endif
+#endif
 	debug("eth_init: fec_probe(bd, %i, %i) @ %08x\n", dev_id, phy_id, addr);
 	bus = fec_get_miibus(base_mii, dev_id);
 	if (!bus)
-- 
1.8.0

